jerry-core/ecma/builtin-objects/typedarray/ecma-builtin-typedarray-prototype.c

```diff
@@ -1628,14 +1628,30 @@ ecma_builtin_typedarray_prototype_slice (ecma_value_t this_arg, /**< this argume
     return ECMA_VALUE_ERROR;
   }
 
-  JERRY_ASSERT (new_typedarray_info.offset == 0);
-
   src_buffer_p += relative_start << info_p->shift;
 
   if (info_p->id == new_typedarray_info.id)
   {
     // 22.2.3.23. Step 22. h-i.
-    memcpy (dst_buffer_p, src_buffer_p, count << info_p->shift);
+
+    if (JERRY_LIKELY (new_typedarray_info.offset == 0))
+    {
+      memcpy (dst_buffer_p, src_buffer_p, count << info_p->shift);
+    }
+    else if (count >= new_typedarray_info.offset)
```

Comment:
What if count is less than the target offset? I don't think this condition is correct.

```
var buf = new ArrayBuffer(10);
var a1 = new Int8Array(buf, 0, 5);
a1.fill(1);
a1.constructor = {
    [Symbol.species]: function (len) {
        return new Int8Array(buf, 5, 5);
    }
};
var a2 = a1.slice(2,4);
res = new Int8Array(buf, 0, 10);
```

Also, I don't think checking the offset is the right way, we need to check whether the two typedarrays use the same arraybuffer instead, and only then fall back to copying element-by-element.